src/receivers/SocketModeFunctions.ts

```diff
@@ -0,0 +1,38 @@
+/* eslint-disable import/prefer-default-export */
+import type { Logger } from '@slack/logger';
+import { CodedError, ErrorCode } from '../errors';
+import { ReceiverEvent } from '../types';
+
+export class SocketModeFunctions {
+  // ------------------------------------------
+  // Error handlers for event processing
+  // ------------------------------------------
+
+  // The default processEventErrorHandler implementation:
+  // Developers can customize this behavior by passing processEventErrorHandler to the constructor
+  public static async defaultProcessEventErrorHandler(
+    args: SocketModeReceiverProcessEventErrorHandlerArgs,
+  ): Promise<boolean> {
+    const { error, logger, event } = args;
+    // TODO: more details like envelop_id, payload type etc. here
+    // To make them available, we need to enhance underlying SocketModeClient
+    // to return more properties to 'slack_event' listeners
+    logger.error('An unhandled error occurred while Bolt processed an event');
```

Comment:
Maybe we can provide a bit more detail in this error message, like which event triggered the error, and, in my humble opinion, the error details should be provided in the noisy `logger.error` message. Otherwise, as a developer having this error occur, I would be frustrated not having any details and not knowing which event triggered a problem and what the problem was.

Reply:
@filmaj This is soooo true! When I started working on this, I tried to add some meaningful information in the log message. However, printing the `body` looked too much for me (the JSON data is not small). Thus, I didn't change the log. Do you think we should remove this error logging until we can add envelope id or something meaningful? If so, I think it's totally fair to remove this error logging at this point.

Reply:
@filmaj Ah, this is nice! Let me merge the suggestion. Thank you! 🙌